app/core/advanced_cache.py

```
@@ -0,0 +1,227 @@
+        # Try L2 cache if available
+        if self.l2_cache is not None:
+            value = self.l2_cache.get(key)
+            if value is not None:
+                # Populate L1 cache
+                self.l1_cache.put(key, value)
+                self.warmer.record_access(key)
+                return value
+                
+        return None
+        
+    def put(self, key: str, value: Any, ttl: Optional[int] = None) -> bool:
```

Ruff (0.8.2)

Dynamically typed expressions (typing.Any) are disallowed in `value`

(ANN401)